lib/logmsg/logmsg.c

```diff
@@ -923,6 +923,41 @@ log_msg_sdata_append_escaped(GString *result, const gchar *sstr, gssize len)
     }
 }
 
+static const gchar *
+find_sd_id_in_key(gint sd_id_len, const gchar *sdata_elem, const gchar *sdata_name,
+                  gint sdata_name_len)
+{
+  if (sd_id_len)
+    {
+      const gchar *sd_id_sep = sdata_elem + sd_id_len;
+      if (sd_id_sep - sdata_name == sdata_name_len)
+        {
+          /* Standalone sdata e.g. [[UserData.updatelist@18372.4]] */
+          sd_id_sep = NULL;
+        }
+      else
+        {
+          g_assert((sd_id_sep - sdata_name < sdata_name_len) && *sd_id_sep == '.');
+        }
+
+      return sd_id_sep;
+    }
+
+  /* The SD-ID has two format, one with enterpriseId:
+   * SD-ID : <key>@<enterprise-id>[.<sub-id>]*.<sd-name> = <sd-value>
+   * Both <enterprise-id> and <sub-id> must be a number.
+   */
+  const gchar *at = memchr(sdata_name + 7, '@', sdata_name_len - 7);
```

Comment:
**minor** comment: there is a constant for ".SDATA." length: https://github.com/syslog-ng/syslog-ng/blob/2d594f664a5d8c01081c96c966ac3d05ef8675d4/lib/logmsg/logmsg.c#L209

Reply:
fixed.